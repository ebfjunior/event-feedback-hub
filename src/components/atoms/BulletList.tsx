import React from 'react';

export type BulletListProps = {
  items?: string[] | null;
  maxItems?: number;
  placeholder?: string;
  className?: string;
};

export function BulletList({ items, maxItems = 5, placeholder = '—', className }: BulletListProps) {
  const safeItems = (items && items.length > 0 ? items : [placeholder]).slice(0, maxItems);
  return (
    <ul className={className}>
      {safeItems.map((item, idx) => (
        <li key={idx} className="flex gap-2">
          <span className="select-none">•</span>
          <span>{item}</span>
        </li>
      ))}
    </ul>
  );
}


