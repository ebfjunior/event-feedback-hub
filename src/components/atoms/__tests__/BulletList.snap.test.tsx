import React from 'react';
import { render } from '@testing-library/react';
import { BulletList } from '../BulletList';

describe('BulletList (snapshot)', () => {
	it('renders with items', () => {
		const { container } = render(<BulletList items={['a', 'b', 'c']} />);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders placeholder when empty', () => {
		const { container } = render(<BulletList items={[]} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});


