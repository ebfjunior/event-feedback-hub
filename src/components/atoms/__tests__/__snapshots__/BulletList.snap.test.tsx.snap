// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BulletList (snapshot) > renders placeholder when empty 1`] = `
<ul>
  <li
    class="flex gap-2"
  >
    <span
      class="select-none"
    >
      •
    </span>
    <span>
      —
    </span>
  </li>
</ul>
`;

exports[`BulletList (snapshot) > renders with items 1`] = `
<ul>
  <li
    class="flex gap-2"
  >
    <span
      class="select-none"
    >
      •
    </span>
    <span>
      a
    </span>
  </li>
  <li
    class="flex gap-2"
  >
    <span
      class="select-none"
    >
      •
    </span>
    <span>
      b
    </span>
  </li>
  <li
    class="flex gap-2"
  >
    <span
      class="select-none"
    >
      •
    </span>
    <span>
      c
    </span>
  </li>
</ul>
`;
